File: solutions/go/difference-of-squares/2/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
	val := n * (n + 1) / 2
	return val * val
}

func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: solutions/fsharp/difference-of-squares/1/DifferenceOfSquares.fs
module DifferenceOfSquares

let squareOfSum (number: int): int = 
    let res = (number * (number + 1) / 2)
    res * res

let sumOfSquares (number: int): int = 
    number * (number + 1) * (2 * number + 1) / 6

let differenceOfSquares (number: int): int = 
    squareOfSum number - sumOfSquares number